device/sycl/src/clusterization/cluster_counting.sycl

```diff
@@ -73,13 +92,13 @@ void cluster_counting(
                         device_cluster_prefix_sum[module_idx];
 
                     // Vector to fill in with the sizes of each cluster
-                    vecmem::device_vector<unsigned int> device_cluster_sizes(
+                    vecmem::device_vector<std::size_t> device_cluster_sizes(
```

Comment:
Wait... does this really work? :confused:

While I prefer to use `std::size_t` for size-type variables as much as possible, `unsigned int` should rather be used for any device-side atomic operations. Since those operations definitely exist on all modern GPUs for `unsigned int`. But not necessarily for `unsigned long` (i.e. `std::size_t`).

Reply:
Alright, I changed it back to unsigned int